packages/relay-compiler/language/javascript/RelayFlowGenerator.js

```diff
@@ -414,6 +409,13 @@ function getFragmentImports(state: State) {
           // TODO(T22653277) support non-haste environments when importing
           // fragments
           imports.push(importTypes([refTypeName], usedFragment + '.graphql'));
+        } else if (
+          state.useSingleArtifactDirectory &&
+          state.existingFragmentNames.has(usedFragment)
+        ) {
+          imports.push(
+            importTypes([refTypeName], './' + usedFragment + '.graphql'),
```

Comment:
hmmm, what exactly are the effects of this?

Reply:
That’s what [my comment above](https://github.com/facebook/relay/pull/2293#issuecomment-364661877) was about. Previously, only users of Haste _and_ those that forked the compiler bin to override the hardcoded disablement of Haste in the OSS compiler bin would be able to benefit from fragment reference type checking, which requires the artefact to import fragment reference types from other artefacts. The assumption is that this was the case because otherwise imports would break if files got moved around. As a non-Haste solution, the alternative is to store all artefacts in the same directory, in which case imports will always be stable and siblings of the current artefact file.

You can see the result here https://github.com/alloy/relay-examples/blob/f8105c243bf476ff0bb0e24de06469ab8e4d62e7/todo/js/__relay_artifacts__/TodoList_viewer.graphql.js#L11-L12

Reply:
I realise that I did not add a test for this, so doing so now.